walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +358,59 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+  
+  @Test
+  public void optionalAttendeesWithGaps() {
+    // Have each all optional attendees have different events. We should see two options because each person has
+    // split the restricted times.
+    //
+    // Events  :       |--A--|     |--B--|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+      new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+      new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request =
+      new MeetingRequest(Arrays.asList(), DURATION_30_MINUTES);
+      request.addOptionalAttendee(PERSON_A);
+      request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+      Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+      TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+      TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
```

Comment:
I think this is a bit more readable as:

```java
Arrays.asList(
  TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
  TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
  TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true)
);
```

Reply:
formatted the test functions with better spacing